### ISSUE TITLE
Accessibility Best Practices for Lottie Animations (SVG Renderer)

### ISSUE THREAD

**Issue by bakztfuture** [airbnb/lottie-web#1935]
Are there any best practices for lottie animations?  I'm using the `SVG` renderer, ideally, I'd like to assign an `alt` tag to the images or SVG elements which Lottie ends up using for the animation on the webpage (instead of an alt tag on its parent div).  

Just curious how others have met accessibility requirements while using lottie animations in the past, couldn't really find any examples online.

**Comment by terenceodonoghue** [comment]
I don't speak for the authors, but my personal inclination would be to use a `role="presentation"` on the parent div if the animation doesn't convey anything meaningful to a screen reader, or `aria-label` if it does.

**Comment by grafst** [comment]
So how should i label my animations for blind users then, @woutthielemans? 😄

**Comment by agarzola** [comment]
I wonder if adding a `role="img"` to the `lottie-player` element (or its parent) would be appropriate. That would make the use of `aria-label` valid (and `aria-labeledby` and `aria-describedby`). MDN [recommends this for SVGs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role#svg_and_roleimg):

> If you are using embedded SVG images in your page, it is a good idea to set `role="img"` on the outer [<svg>](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/svg) element and give it a label. This will cause screen readers to just consider it as a single entity and describe it using the label, rather than trying to read out all the child nodes

Note that, according to that MDN page, browsers automatically apply `role="presentational"` to all descendants of an element with role `img`. So if you do apply `role="img"` to the parent element, be sure it doesn’t also contain other elements that should be part of the accessibility tree.

**Comment by bauerbua** [comment]
LottieFiles provide a [guide on WCAG 2.1 compliance](https://developers.lottiefiles.com/docs/resources/wcag/). Setting `role="img"` and using a descriptive `aria-label` seems to be the way to go for basic animations as @agarzola pointed out.

**Comment by macroantu** [comment]
This worked for me 
If the animation is purely decorative, you can hide the generated SVG but still label the parent:

`    animation.addEventListener('DOMLoaded', function() {
      const svg = document.querySelector('#lottie-animation svg');
      if (svg) {
        svg.setAttribute('aria-hidden', 'true');
        svg.removeAttribute('aria-label');
      }
    });`
